### PR TITLE
Review of rdkit.Chem.MolKey

### PR DESCRIPTION
**Pull request by gedeck**
InchiInfo.py contained unittest code. I moved this one to UnitTestMolKey to separate the tests from the code. The file UnitTestMolKey also executes the MolKey doctests and has a few more tests to increase coverage of MolKey.py.

### CODE REVIEW

**Comment by gedeck** [comment]
There is a branch `rdkit:MolKey_Aug2015` that contains a modification to the `struchk_init` string. The current code has a `\n` at the end, while the branch removes this. Other changes are formatting only. It seems to me that the pyAvalonTools code that is called with the string ignores the new line.

**Comment by greglandrum** [comment]
About the newline: I think there was a version of the AvalonTools where this was necessary on at least one operating system. As long as what's there works we can/should stick with it.

File: rdkit/Chem/MolKey/MolKey.py [hunk 1]
```diff
@@ -410,7 +412,8 @@ def GetKeyForCTAB(ctab, stereo_info=None, stereo_comment=None, logger=None):
   try:
     err, inchi, fixed_mol = GetInchiForCTAB(ctab)
   except BadMoleculeException:
-    logger.warn('Corrupt molecule substituting no-struct: --->\n{0}\n<----'.format(ctab))
+    msg = u'Corrupt molecule substituting no-struct: --->\n{0}\n<----'.format(ctab)
```
Comment (greglandrum) [inline rdkit/Chem/MolKey/MolKey.py]: Is the u actually necessary here?

File: rdkit/Chem/MolKey/UnitTestMolKey.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
+# $Id$
```
Comment (greglandrum) [inline rdkit/Chem/MolKey/UnitTestMolKey.py]: please no svn droppings in the code. ;-)

File: rdkit/Chem/MolKey/UnitTestMolKey.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+#
+#  Copyright (C) 2006  greg Landrum
```
Comment (greglandrum) [inline rdkit/Chem/MolKey/UnitTestMolKey.py]: The 2006 is dead wrong here. Probably best to just remove this line entirely.

File: rdkit/Chem/MolKey/MolKey.py (hunk 1 above)
Comment (gedeck) [inline rdkit/Chem/MolKey/MolKey.py]: That one drove me mad. Without the `u`, the (new) test that raises the Exception failed with a unicode error on Python 2.7. This works, but I honestly don't understand it why it was necessary.

File: rdkit/Chem/MolKey/UnitTestMolKey.py (hunk 2 above)
Comment (gedeck) [inline rdkit/Chem/MolKey/UnitTestMolKey.py]: Let's raise an issue and remove them on all code.

**Pull request merged**